xcodeproj/internal/xcodeprojinfo.bzl

```diff
@@ -289,12 +289,40 @@ def _skip_target(
         label_str = bazel_labels.normalize_label(info.xcode_target.label)
         package_label_str = label_str.split(":")[0]
 
+        # Important notes:
+        #
+        # 1. This relies on implementation details of
+        # https://github.com/bazelbuild/rules_apple/blob/master/apple/internal/testing/apple_test_assembler.bzl#L100-L125
+        # 2. Since `target_skip_type` is `skip_type.apple_test_bundle`
+        # we can assume that `runner` is present as it's a required attribute
+        # 3. The `.replace` below is safe even if `runner` is the
+        # default runner `ios_default_runner` since it's a no-op in that case.
+        #
+        # Removes `runner` from `target.label.name` to create a replacement
+        # label that works in all scenarios. For context, this is why
+        # other approaches won't work:
+        #
+        # 1. Simply use `target.label.name`
+        #
         # As of https://github.com/bazelbuild/rules_apple/pull/1948
         # `bundle_name` can be used to name the bundle instead of the
-        # target name. Because of that we use `ctx.rule.attr.generator_name`
-        # here to ensure this is always a real target label.
+        # target name and `bundle_name` is not accesible in this context.
+        #
+        # 2. Use `ctx.rule.attr.generator_name`
+        #
+        # In many scenarios `generator_name` would be enough since it holds the
+        # name of the test rule generating it but if the test
+        # targets are wrapped in macros then `generator_name` can be anything
+        #
+        # Example:
+        #
+        # `iOSAppObjCUnitTestSuite_iPhone-13-Pro__16.2` => `iOSAppObjCUnitTestSuite`
+        #
```

Comment:
@mattrobmattrob @luispadron @brentleyjones this summarizes the approach and my reasoning, let me know if it makes sense and if it addresses the other review comments or not